src/lib/helpers.js

```diff
@@ -52,6 +53,25 @@ const calculateBucketId = time => {
   return Math.floor(timeNumber / BUCKET_DURATION);
 };
 
+const getCoverTrancheAllocations = (cover, poolId, now) => {
+  const packedTrancheAllocations = cover.poolAllocations.find(p => p.poolId === poolId)?.packedTrancheAllocations;
+  if (!packedTrancheAllocations) {
+    return [];
+  }
+
+  const bitmask32 = ethers.BigNumber.from('0xFFFFFFFF');
+  const firstActiveTrancheId = calculateTrancheId(now);
+  const offset = firstActiveTrancheId - cover.coverData.start / TRANCHE_DURATION;
+
+  const coverTrancheAllocations = [];
+  for (let i = offset; i < MAX_ACTIVE_TRANCHES; i++) {
+    const allocation = packedTrancheAllocations.shr(i * 32).and(bitmask32);
+    coverTrancheAllocations.push(allocation);
```

Comment:
The indexes here are not preserved, this will lead to inaccurate results when subtracting from incorrect tranches. We need to subtract the amount from the exact tranche it was allocated to because the allocation could be greater than the capacity (due to posible burns) and we need to know exactly if we freed up anything in there or not (and the tranche is important, because it may or may not be usable for the new cover).